### PR TITLE
Add preferences for analytics

### LINKED ISSUE #821: Add feature flag for analytics
## Description

To kick off development of analytics, a feature flag `analytics` must be created, which will be set to the following statuses:

- enabled in dev
- switchable in staging
- disabled in prod

## Additional context

RFC: https://github.com/WordPress/openverse/blob/3fa9f03937d2af356535f158f7f80b20a6bd79d2/rfcs/analytics/20221006-implementation_plan_frontend.md

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #821 by @dhruvkb

## Description

This PR
- adds a switch view to the checkbox based on the WordPress design language
- removes dummy example feature flags
- adds feature flag for analytics
- updates the preferences UI to match mockups from #791
- adds documentation for the feature flag system

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
The first thing I want to mention is that the preferences page is still internal and not for public use. What I meant by the UI updated to match the mockups is only regarding this segment:

Screenshot 2023-03-06 at 9 42 15 PM

Francisco's mockups in #791 suggest a popover for settings and this area matches the contents of that popup.

So while you're raising some very good points in the first half of your review, they don't really apply here because making this page public is not the plan here.

## Bugs, other thoughts

- The code block renders just fine for me. I tested it on Firefox and Chrome and it renders the complete data for me.
- I can't remember what that section is exactly, which is a pretty good sign that it can just be removed altogether 😄.
- The switches have a white knob when enabled and match the WP DL 100%. I tested it on Firefox and Chrome and it renders the white knob for me.
 
It is very weird that you're seeing UI bugs which are rendering fine for me. Screenshot attached for reference:

image

File: frontend/src/pages/preferences.vue [hunk 1]
```diff
@@ -2,25 +2,65 @@
   <VContentPage>
     <h1>{{ $t("pref-page.title") }}</h1>
 
-    <div v-for="isSwitchable in [false, true]" :key="isSwitchable">
-      <h2>
+    <!-- TODO: Extract this to preferences modal. -->
```
Comment (dhruvkb) [inline frontend/src/pages/preferences.vue:5]: @zackkrida only this bit is what will be made public later on.

Comment (zackkrida) [reply frontend/src/pages/preferences.vue:5]: Definitely! I understand now. I incorrectly assumed this PR was suggesting the page be used publicly. We can save my suggestions for if (and when) that day comes. Clever to implement some of the early design work for preferences in this PR.

**Review by zackkrida — APPROVED** [review]
oh god, I had JS disabled on the page when reviewing the first time 🤦 . 

I tested the a11y of the checkbox and I think it's good. Especially considering this is not intended to be a public page, I think we can save additional auditing of that component for the future. LGTM.

**Comment by dhruvkb** [comment]
Upped the priority to high because I'd like to be able to use this work in #862.

File: frontend/feat/feature-flags.json [hunk 2]
```diff
@@ -15,18 +15,18 @@
       },
       "description": "Toggle the new header, including recent searches."
```
Comment (obulat) [inline frontend/feat/feature-flags.json:16]: Can we remove the `new_header` flag, too?

File: frontend/tailwind.config.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const { Z_INDICES } = require("./src/constants/z-indices")
 
 module.exports = {
   content: [
-    "src/**/*.{vue,js,jsx,ts,tsx,mdx}",
+    "./src/**/*.{vue,js,jsx,ts,tsx,mdx}",
```
Comment (obulat) [inline frontend/tailwind.config.js:9]: Was something not working when we had no './' here?

**Review by obulat — APPROVED** [review]
Looks beautiful! I have a question and a suggestion inline, but they are not blocking.

File: frontend/tailwind.config.js (hunk 3 above)
Comment (dhruvkb) [reply frontend/tailwind.config.js:9]: I don't remember why I added this. Considering that, I'll drop this change.

File: frontend/feat/feature-flags.json (hunk 2 above)
Comment (dhruvkb) [reply frontend/feat/feature-flags.json]: Done in ff12982.

**Pull request merged**